sql/ddl_api.sql

```diff
@@ -308,3 +308,152 @@ BEGIN
     PERFORM _timescaledb_internal.attach_tablespace(hypertable_id, tablespace);
 END
 $BODY$;
+
+
+CREATE OR REPLACE FUNCTION reindex_indexes(
+    index_oids OID[],
+    verbose   BOOLEAN = FALSE,
+    recreate  BOOLEAN = FALSE
+)
+    RETURNS VOID LANGUAGE PLPGSQL VOLATILE AS
+$BODY$
+DECLARE
+    new_index_oids OID[];
+    index_oid OID;
+    new_index_oid OID;
+    verbose_mod TEXT = '';
+    index_name NAME;
+    index_schema_name NAME;
+    recreate_state_info RECORD;
+BEGIN
+    FOR index_oid IN SELECT unnest(index_oids)
+    LOOP
+        SELECT c.relname, n.nspname INTO STRICT index_name, index_schema_name
+        FROM pg_class c
+        INNER JOIN pg_namespace n ON (n.OID = c.relnamespace)
+        WHERE c.OID = index_oid;
+
+        IF recreate THEN
+            -- the recreate phase is split into 2 parts a (1) CREATE INDEX followed by  (2) DROP INDEX + RENAME INDEX.
+            -- you want to do phase 1 on all indexes before starting phase 2 since phase 2 takes heavy locks but is quick.
+            SELECT _timescaledb_internal.chunk_index_recreate_create(index_oid)
+            INTO new_index_oid;
+
+            IF "verbose" = true THEN
+                RAISE INFO 'a new index was created for index "%"', index_name;
+            END IF;
+            new_index_oids :=  new_index_oids || new_index_oid;
+        ELSE
+            IF "verbose" THEN
+                verbose_mod := '(VERBOSE)';
+            END IF;
+            EXECUTE FORMAT($$ REINDEX %s INDEX %I.%I $$, verbose_mod, index_schema_name, index_name);
+        END IF;
+    END LOOP;
+
+    IF recreate THEN
+        FOR recreate_state_info IN SELECT unnest(index_oids) AS old_oid,  unnest(new_index_oids) AS new_oid
+        LOOP
+            -- phase 2 of recreate index
+            SELECT relname INTO STRICT index_name FROM pg_class WHERE OID = recreate_state_info.old_oid;
+            PERFORM _timescaledb_internal.chunk_index_recreate_rename(recreate_state_info.old_oid, recreate_state_info.new_oid);
+            IF "verbose" = true THEN
+                RAISE INFO 'index "%" was renamed to use the new index', index_name;
+            END IF;
+        END LOOP;
+    END IF;
+END
+$BODY$;
+
+--reindex only some chunks of a hypertable
+--main_table is the hypertable
+--index_oid is the index on a hypertable to reindex (NULL for all)
+--verbose is a flag to indicate whether to output a notice for each index processed
+--recreate is a flag to force a CREATE INDEX followed by a DROP INDEX instead of a REINDEX.
+--time_column the name of the time column.
+--from_time is a lower bound for the time filter on a chunk
+--to_time is a upper bound for the time filter on a chunk
+----A reindex locks out reads during the procedure if those reads use the index.
+----A recreate does not lock reads but may use more disk space.
+CREATE OR REPLACE FUNCTION reindex_index_chunk(
+    main_table REGCLASS,
+    index_oid REGCLASS,
```

Comment:
Why do you need both table and index REGCLASS here? Can't you just give one relation regclass arg, and then figure out whether it is a table or index? In case of index, you get table from `pg_index`.

Reply:
changed